satpy/readers/seviri_l1b_native.py

```diff
@@ -193,10 +193,27 @@ def __init__(self, filename, filename_info, filetype_info,
         # Available channels are known only after the header has been read
         self.header_type = get_native_header(has_archive_header(self.filename))
         self._read_header()
-        self.dask_array = da.from_array(self._get_memmap(), chunks=(CHUNK_SIZE,))
+        self._make_dask_array_with_map_blocks()
         self._read_trailer()
         self.image_boundaries = ImageBoundaries(self.header, self.trailer, self.mda)
 
+    def _make_dask_array_with_map_blocks(self):
+        """Makes the dask array using the ``da.map_blocks()`` functionality."""
+        dtype = self._get_data_dtype()
+        chunks = da.core.normalize_chunks(
+            "auto",
+            shape=(self.mda["number_of_lines"],),
+            dtype=dtype)
+        self.dask_array = da.map_blocks(
```

Comment:
Does `.dask_array` need to be public? Can it get a `_` prefix (`self._dask_array`)? Oh I see, it was like this before. Maybe not something that *needs* to be done.

Reply:
Done!